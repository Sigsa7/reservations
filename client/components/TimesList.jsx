import React from 'react';
import PropTypes from 'prop-types';

const TimesList = ({ bookings }) => (
  <div>
    {JSON.stringify(bookings)}
  </div>
);

TimesList.propTypes = {
  bookings: PropTypes.arrayOf(PropTypes.shape({
    booking_time: PropTypes.any,
    party_qty: PropTypes.number,
  })).isRequired,
};

export default TimesList;
