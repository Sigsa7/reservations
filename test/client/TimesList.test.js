import React from 'react';
import renderer from 'react-test-renderer';

import TimesList from '../../client/components/TimesList';

describe('TimesList component', () => {
  const sampleBookings = [{
    booking_time: null,
    party_qty: 0,
  }];

  it('renders with no errors when valid bookings array passed in', () => {
    const component = renderer.create(<TimesList bookings={sampleBookings} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
