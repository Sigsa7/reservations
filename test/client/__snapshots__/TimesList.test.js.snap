// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimesList component renders with no errors when valid bookings array passed in 1`] = `
<div>
  [{"booking_time":null,"party_qty":0}]
</div>
`;

exports[`TimesList component renders with no errors with valid bookings array 1`] = `
<div>
  [{"booking_time":null,"party_qty":0}]
</div>
`;
